File: src/main/java/com/yui/models/NFT.java
package com.yui.models;

import java.sql.Timestamp;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name="nfts")
public class NFT {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;
    private String token;
    
    @ManyToOne
    @JoinColumn(name="artwork_id")
    private Artwork artwork;

    private Timestamp createdAt;
}

File: src/entity/nft.ts
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { Artwork } from '.';

@Entity()
export class NFT {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    token: string;

    @ManyToOne(() => Artwork, (artwork) => artwork.nfts)
    artwork: Artwork;

    @Column()
    createdAt: Date;
}
